Marlin/Marlin_main.cpp

```diff
@@ -7598,15 +7598,27 @@ inline void gcode_M907() {
  *
  *   S<bool>  Turn case light on or off
  *   P<byte>  Set case light brightness (PWM pin required)
+ *
+ *   S and P will both work as intended no matter if a digital or PWM pin
+ *      S >= 128 will turn on a digital pin
  */
 inline void gcode_M355() {
   #if HAS_CASE_LIGHT
-    if (code_seen('P')) case_light_brightness = code_value_byte();
-    if (code_seen('S')) case_light_on = code_value_bool();
+    if (code_seen('P')) {
+      case_light_brightness = code_value_byte();
+      case_light_brightness > 127 ? case_light_on = 1 : case_light_on = 0;
+    }
+    if (code_seen('S')) {
+      case_light_on = code_value_bool();
+      case_light_on ? case_light_brightness = 255 : case_light_brightness = 0;
+    }
```

Comment:
Personally I'm against this.  If I specify P and S at the same time, S will always override P.  On top of that, why turn off the case light with brightness below 127?  I'll turn mine down to 64 or so to see the printer late at night without lighting up the whole room.